dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/Engine.scala

```diff
@@ -86,18 +86,18 @@ class Engine(context: EngineContext) {
   /** Submit tasks to a worker pool, solving them in parallel. Upon receiving results for a task, new tasks are
     * submitted accordingly. Once no more tasks can be created, the list of results is returned.
     */
-  private def solveTasks(tasks: List[ReachableByTask], sources: Set[CfgNode]): List[ReachableByResult] = {
+  private def solveTasks(tasks: List[ReachableByTask], sources: Set[CfgNode]): Vector[ReachableByResult] = {
     var completedResults = List[ReachableByResult]()
 
     def handleSummary(taskSummary: TaskSummary): Unit = {
+      val newTasks = taskSummary.followupTasks
+      newTasks.par.foreach(task => submitTask(task, sources))
```

Comment:
... and this line. Confirmed: submission of tasks in parallel makes results flaky.

Reply:
This is done.